taichi/transforms/offload.cpp

```diff
@@ -414,9 +482,10 @@ OffloadedResult offload(IRNode *root) {
     PromoteIntermediate::run(root, result.local_to_global_offset);
     PromoteLocals::run(root, result.local_to_global_offset);
   }
-  irpass::insert_gc(root);
-  irpass::typecheck(root);
-  irpass::re_id(root);
+  insert_gc(root);
+  AssociateContinueScope::run(root);
```

Comment:
I put this pass here so that all backends get this step for free before they are integrated with #700 . Do you think we should create a new pass for this in the future?

Reply:
Ack, i plan to make it a new pass once we all switch to this.. https://github.com/taichi-dev/taichi/blob/213a3c7c02823a95d67ec2c9cf562c7f0389746c/taichi/transforms/compile_to_offloads.cpp#L7